code/modules/projectiles/guns/energy.dm

```diff
@@ -593,4 +597,367 @@
 	var/obj/screen/ammo/A = user.hud_used.ammo //The ammo HUD
 	A.update_hud(user)
 
+/obj/item/weapon/gun/energy/lasgun/lasrifle/update_item_state(mob/user)// lasgun code thing i don't want so redfined
+	. = item_state
+	item_state = "[initial(icon_state)][flags_item & WIELDED ? "_w" : ""]"
+	if(. != item_state && ishuman(user))
+		var/mob/living/carbon/human/human_user = user
+		if(src == human_user.l_hand)
+			human_user.update_inv_l_hand()
+		else if (src == human_user.r_hand)
+			human_user.update_inv_r_hand()
+
```

Comment:
Use early returns
explain this code olease

Reply:
Parent lasgun is shitecode and needs this in order to display the correct wielded sprite